connectors/source-mysql/src/main/java/com/linkall/source/mysql/MySqlConfig.java

```diff
@@ -0,0 +1,152 @@
+import io.debezium.storage.file.history.FileSchemaHistory;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.json.JsonConverterConfig;
+import org.apache.kafka.connect.storage.Converter;
+
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class MySqlConfig extends DebeziumConfig {
+  private static final Set<String> systemTable = new HashSet(Arrays.asList("information_schema", "mysql", "performance_schema", "sys"));
+
+  @JsonProperty("db_config")
+  private DbConfig dbConfig;
+
+  @JsonProperty("binlog_offset")
+  private BinlogOffset binlogOffset;
+
+  @JsonProperty("db_history_file")
+  private String dbHistoryFile;
+
+  @JsonProperty("include_tables")
+  private String[] includeTables;
+
+  @JsonProperty("exclude_tables")
+  private String[] excludeTables;
+
+
+  @Override
+  public Class<?> secretClass() {
+    return DbConfig.class;
+  }
+
+  @Override
+  protected Properties getDebeziumProperties() {
+    final Properties props = new Properties();
+
+    // debezium engine configuration
+    props.setProperty("connector.class", MySqlConnector.class.getCanonicalName());
+    // snapshot config
+    props.setProperty("snapshot.mode", "initial");
+    // DO NOT include schema change, e.g. DDL
+    props.setProperty("include.schema.changes", "false");
+    // disable tombstones
+    props.setProperty("tombstones.on.delete", "false");
+
+    if (binlogOffset != null) {
+      Converter valueConverter = new JsonConverter();
+      Map<String, Object> valueConfigs = new HashMap<>();
+      valueConfigs.put(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, false);
+      valueConverter.configure(valueConfigs, false);
+      byte[] offsetValue =
+          valueConverter.fromConnectData(dbConfig.getDatabase(), null, binlogOffset);
+      props.setProperty(
+          KvStoreOffsetBackingStore.OFFSET_CONFIG_VALUE,
+          new String(offsetValue, StandardCharsets.UTF_8));
+    }
+
+    props.setProperty("offset.flush.interval.ms", "1000");
+
+    // history
+    props.setProperty("schema.history.internal", FileSchemaHistory.class.getCanonicalName());
+    props.setProperty("schema.history.internal.file.filename", dbHistoryFile);
+
+    // https://debezium.io/documentation/reference/configuration/avro.html
+    props.setProperty("key.converter.schemas.enable", "false");
+    props.setProperty("value.converter.schemas.enable", "false");
+
+    // debezium names
+    props.setProperty("name", dbConfig.getDatabase());
+    props.setProperty("topic.prefix", dbConfig.getDatabase());
+    props.setProperty("database.server.id", String.valueOf(System.currentTimeMillis()));
+
+    // db connection configuration
+    props.setProperty("database.hostname", dbConfig.getHost());
+    props.setProperty("database.port", String.valueOf(dbConfig.getPort()));
+    props.setProperty("database.dbname", dbConfig.getDatabase());
+    if (dbConfig.getUsername() != null && dbConfig.getUsername() != "")
```

Comment:
`dbConfig != null && dbConfig.getUsername() != ""`?

Reply:
remove it